### PR TITLE
Submit Application as a Business

### PR DESCRIPTION
**Pull request by dimak1**
*Issue:*

- bcgov/entity/issues/24023

*Description of changes:*
- Submit the Application in my Business name
- Add dynamic validation
- Refactor components, tech debt cleanup
- Unit tests

Screenshot 2024-11-04 at 15 54 24

Screenshot 2024-11-04 at 15 54 19

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

File: strr-web/stores/strr.ts [hunk 1]
```diff
@@ -92,19 +89,25 @@ const optionalOrEmptyString = z
   .string()
   .optional()
   .transform(e => (e === '' ? undefined : e))
-const requiredNonEmptyString = z.string().refine(e => e !== '', 'Field cannot be empty')
+const requiredNonEmptyString = z
+  .string()
+  .min(1, { message: 'Required' })
+  // .refine(e => e !== '', 'Field cannot be empty')
```
Comment (dimak1) [inline strr-web/stores/strr.ts:95]: Left comments as a reference or if we need to revert

File: strr-web/components/bcros/form-section/contact-information/Form.vue [hunk 2]
```diff
@@ -200,16 +234,69 @@ onMounted(() => {
...
+  if (activeAddressField.value === 'primaryContactAddress') {
+    return formState.primaryContact
+  } else if (activeAddressField.value === 'secondaryContactAddress') {
+    return formState.secondaryContact
+  } else {
+    return formState.propertyDetails
+  }
+}
+
+// reset errors for Primary Contact
```
Comment (dimak1) [inline strr-web/components/bcros/form-section/contact-information/Form.vue:247]: Validation below needs to be reworked globally, but for now kept it for visibility

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-267-m6ngvnla.web.app

**Comment by rstens** [comment]
Why are there 2 Preferred name fields?
image

**Comment by dimak1** [comment]
@rstens thanks for catching it, I reviewed the app/figma and added more fixes. Should be good now.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-267-m6ngvnla.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-267-m6ngvnla.web.app

File: strr-web/components/bcros/form-section/BusinessDetails.vue [hunk 3]
```diff
@@ -1,26 +1,28 @@
...
+    <BcrosFormSection :title="t('common.formLabels.businessDetails')">
+      <div class="flex flex-row justify-between w-full mb-10 m:flex-col m:mb-4">
         <UFormGroup name="businessName" class="flex-grow" :error="errors.businessLegalName">
           <UInput
             v-model="businessName"
             type="text"
-            :placeholder="t('createAccount.propertyManagerForm.businessName')"
-            data-test-id="property-manager-business-name-input"
+            :placeholder="isBusinessNameRequired
+              ? t('common.formLabels.businessLegalName')
```
Comment (shaangill025) [inline strr-web/components/bcros/form-section/BusinessDetails.vue:10]: @dimak1 Re validation - are you enforcing when it is a required field.

**Review by shaangill025 — APPROVED** [review]

File: strr-web/components/bcros/form-section/BusinessDetails.vue (hunk 3 above)
Comment (dimak1) [reply strr-web/components/bcros/form-section/BusinessDetails.vue:10]: When the `contactType` is Business, this field is required. So I created this prop to force the input.

Comment (dimak1) [reply strr-web/components/bcros/form-section/BusinessDetails.vue:10]: All good, can we resolve it?

**Pull request merged**